WordPress/WordPressTest/Services/PostAutoUploadInteractorTests.swift

```diff
@@ -101,16 +101,34 @@ class PostCoordinatorUploadActionUseCaseTests: XCTestCase {
             expect(actualResult).to(equal(expectedCancelableResult))
         }
     }
+
+    func testReturnNothingAsPostActionWhenAttemptLimitIsReached() {
+        let post = createPost(.publish, confirmedAutoUpload: true, autoUploadFailureCount: 3)
+
+        let action = interactor.autoUploadAction(for: post)
+
+        expect(action).to(equal(.nothing))
+    }
+
+    func testReturnUploadAsPostActionWhenAttemptLimitIsNotReached() {
+        let post = createPost(.publish, confirmedAutoUpload: true, autoUploadFailureCount: 2)
+
+        let action = interactor.autoUploadAction(for: post)
+
+        expect(action).to(equal(.upload))
+    }
 }
 
 private extension PostCoordinatorUploadActionUseCaseTests {
     func createPost(_ status: BasePost.Status,
                     remoteStatus: AbstractPostRemoteStatus = .failed,
                     hasRemote: Bool = false,
-                    confirmedAutoUpload: Bool = false) -> Post {
+                    confirmedAutoUpload: Bool = false,
+                    autoUploadFailureCount: Int = 1) -> Post {
```

Comment:
Should we call this as `autoUploadAttemptsCount` too? Or `attemptsCount` sounds concise enough.

Reply:
Done in 6fe1bb8a4b8ac49b7c194a5c550299aa3788d849